### PR TITLE
Fix the CountUpdatedPresetsAfterApplyingPendingPresets utility functi…

### LINKED ISSUE #34778: Need to fix the CountUpdatedPresetsAfterApplyingPendingPresets to return the size of the pending preset list
Need to resolve comment - https://github.com/project-chip/connectedhomeip/pull/34532#discussion_r1693633732

### PR DESCRIPTION
**Pull request by nivi-apple**
…on to return the size of the pending presets list

Fixes: #34778

### CODE REVIEW

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 1]
```diff
@@ -372,16 +347,14 @@ uint8_t CountUpdatedPresetsAfterApplyingPendingPresets(Delegate * delegate)
...
                          err.Format());
             return 0;
         }
         numberOfPendingPresets++;
     }
 
-    // TODO: #34546 - Need to support deletion of presets that are removed from Presets.
-    // This API needs to modify its logic for the deletion case.
-    return static_cast<uint8_t>(numberOfPresets + numberOfPendingPresets - numberOfMatches);
+    return static_cast<uint8_t>(numberOfPendingPresets);
```
Comment (hasty) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:357]: This is already a uint8_t.

Comment (nivi-apple) [reply src/app/clusters/thermostat-server/thermostat-server.cpp]: done.

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 2]
```diff
@@ -321,46 +321,21 @@ bool IsPresetHandlePresentInPresets(Delegate * delegate, const ByteSpan & preset
...
+ * @brief Returns the length of the list of presets if the pending presets were to be applied. The size of the pending presets list
+ *        calculated, after all the constraint checks are done, is the new size of the updated Presets attribute since the pending
+ *        preset list is expected to have all existing presets with or without edits plus new presets.
+ *        This is called before changes are actually applied.
  *
  * @param[in] delegate The delegate to use.
  *
  * @return count of the updated Presets attribute if the pending presets were applied to it. Return 0 for error cases.
  */
 uint8_t CountUpdatedPresetsAfterApplyingPendingPresets(Delegate * delegate)
```
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:333]: How about we just rename this to CountNumberOfPendingPresets?  That's what it does, and that's what the caller cares about.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/clusters/thermostat-server/thermostat-server.cpp (hunk 2 above)
Comment (nivi-apple) [reply src/app/clusters/thermostat-server/thermostat-server.cpp]: done.

**Review by bauerschwan — APPROVED** [review]

**Review by hasty — APPROVED** [review]

**Pull request merged**